# **TikTok Project**
**Course 4 - The Power of Statistics**

You are a data professional at TikTok. The current project is reaching its midpoint; a project proposal, Python coding work, and exploratory data analysis have all been completed.

The team has reviewed the results of the exploratory data analysis and the previous executive summary the team prepared. You received an email from Orion Rainier, Data Scientist at TikTok, with your next assignment: determine and conduct the necessary hypothesis tests and statistical analysis for the TikTok classification project.

A notebook was structured and prepared to help you in this project. Please complete the following questions.


# **Course 4 End-of-course project: Data exploration and hypothesis testing**

In this activity, you will explore the data provided and conduct hypothesis testing.
<br/>

**The purpose** of this project is to demostrate knowledge of how to prepare, create, and analyze hypothesis tests.

**The goal** is to apply descriptive and inferential statistics, probability distributions, and hypothesis testing in Python.
<br/>

*This activity has three parts:*

**Part 1:** Imports and data loading
* What data packages will be necessary for hypothesis testing?

**Part 2:** Conduct hypothesis testing
* How will descriptive statistics help you analyze your data?

* How will you formulate your null hypothesis and alternative hypothesis?

**Part 3:** Communicate insights with stakeholders

* What key business insight(s) emerge from your hypothesis test?

* What business recommendations do you propose based on your results?

<br/>

Follow the instructions and answer the questions below to complete the activity. Then, complete an executive summary using the questions listed on the PACE Strategy Document.

Be sure to complete this activity before moving on. The next course item will provide you with a completed exemplar to compare to your own work.



# **Data exploration and hypothesis testing**

<img src="images/Pace.png" width="100" height="100" align=left>

# **PACE stages**

Throughout these project notebooks, you'll see references to the problem-solving framework PACE. The following notebook components are labeled with the respective PACE stage: Plan, Analyze, Construct, and Execute.

<img src="images/Plan.png" width="100" height="100" align=left>


## **PACE: Plan**

Consider the questions in your PACE Strategy Document and those below to craft your response.

1. What is your research question for this data project? Later on, you will need to formulate the null and alternative hypotheses as the first step of your hypothesis test. Consider your research question now, at the start of this task.

**Answer:**  
Is there a statistically significant difference in the number of views for TikTok videos posted by verified accounts versus unverified accounts?

*Complete the following steps to perform statistical analysis of your data:*

### **Task 1. Imports and Data Loading**

Import packages and libraries needed to compute descriptive statistics and conduct a hypothesis test.

<details>
  <summary><h4><strong>Hint:</strong></h4></summary>

Be sure to import `pandas`, `numpy`, `matplotlib.pyplot`, `seaborn`, and `scipy`.

</details>

In [1]:
# Import packages for data manipulation
import numpy as np
import pandas as pd

# Import packages for data visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Import packages for statistical analysis/hypothesis testing
import statsmodels.api as sm
from scipy import stats

Load the dataset.

**Note:** As shown in this cell, the dataset has been automatically loaded in for you. You do not need to download the .csv file, or provide more code, in order to access the dataset and proceed with this lab. Please continue with this activity by completing the following instructions.

In [2]:
# Load dataset into dataframe
data = pd.read_csv("tiktok_dataset.csv")

<img src="images/Analyze.png" width="100" height="100" align=left>

<img src="images/Construct.png" width="100" height="100" align=left>

## **PACE: Analyze and Construct**

Consider the questions in your PACE Strategy Document and those below to craft your response:
1. Data professionals use descriptive statistics for Exploratory Data Analysis. How can computing descriptive statistics help you learn more about your data in this stage of your analysis?


Descriptive satistics is a good way to view information on large datasets by computing the mean, median, quartiles, etc.

### **Task 2. Data exploration**

Use descriptive statistics to conduct Exploratory Data Analysis (EDA).



<details>
  <summary><h4><strong>Hint:</strong></h4></summary>

Refer back to *Self Review Descriptive Statistics* for this step-by-step proccess.

</details>

Inspect the first five rows of the dataframe.

In [3]:
# Display first few rows
data.head()

# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

In [4]:
# Generate a table of descriptive statistics about the data
data.describe(include='all')

# claim_status      video_id  video_duration_sec  \
count   19382.000000        19084  1.938200e+04        19382.000000   
unique           NaN            2           NaN                 NaN   
top              NaN        claim           NaN                 NaN   
freq             NaN         9608           NaN                 NaN   
mean     9691.500000          NaN  5.627454e+09           32.421732   
std      5595.245794          NaN  2.536440e+09           16.229967   
min         1.000000          NaN  1.234959e+09            5.000000   
25%      4846.250000          NaN  3.430417e+09           18.000000   
50%      9691.500000          NaN  5.618664e+09           32.000000   
75%     14536.750000          NaN  7.843960e+09           47.000000   
max     19382.000000          NaN  9.999873e+09           60.000000   

                                 video_transcription_text verified_status  \
count                                               19084           19382   
unique                                              19012               2   
top     a friend read  in the media a claim that badmi...    not verified   
freq                                                    2           18142   
mean                                                  NaN             NaN   
std                                                   NaN             NaN   
min                                                   NaN             NaN   
25%                                                   NaN             NaN   
50%                                                   NaN             NaN   
75%                                                   NaN             NaN   
max                                                   NaN             NaN   

       author_ban_status  video_view_count  video_like_count  \
count              19382      19084.000000      19084.000000   
unique                 3               NaN               NaN   
top               active               NaN               NaN   
freq               15663               NaN               NaN   
mean                 NaN     254708.558688      84304.636030   
std                  NaN     322893.280814     133420.546814   
min                  NaN         20.000000          0.000000   
25%                  NaN       4942.500000        810.750000   
50%                  NaN       9954.500000       3403.500000   
75%                  NaN     504327.000000     125020.000000   
max                  NaN     999817.000000     657830.000000   

        video_share_count  video_download_count  video_comment_count  
count        19084.000000          19084.000000         19084.000000  
unique                NaN                   NaN                  NaN  
top                   NaN                   NaN                  NaN  
freq                  NaN                   NaN                  NaN  
mean         16735.248323           1049.429627           349.312146  
std          32036.174350           2004.299894           799.638865  
min              0.000000              0.000000             0.000000  
25%            115.000000              7.000000             1.000000  
50%            717.000000             46.000000             9.000000  
75%          18222.000000           1156.250000           292.000000  
max         256130.000000          14994.000000          9599.000000

Check for and handle missing values.

In [5]:
# Check for missing values
data.isna().sum()

#                             0
claim_status                298
video_id                      0
video_duration_sec            0
video_transcription_text    298
verified_status               0
author_ban_status             0
video_view_count            298
video_like_count            298
video_share_count           298
video_download_count        298
video_comment_count         298
dtype: int64

In [6]:
# Drop rows with missing values
data_cleaned = data.dropna(axis=0)

data_cleaned.isna().sum()

#                           0
claim_status                0
video_id                    0
video_duration_sec          0
video_transcription_text    0
verified_status             0
author_ban_status           0
video_view_count            0
video_like_count            0
video_share_count           0
video_download_count        0
video_comment_count         0
dtype: int64

In [7]:
# Display first few rows after handling missing values
data_cleaned.head()

# claim_status    video_id  video_duration_sec  \
0  1        claim  7017666017                  59   
1  2        claim  4014381136                  32   
2  3        claim  9859838091                  31   
3  4        claim  1866847991                  25   
4  5        claim  7105231098                  19   

                            video_transcription_text verified_status  \
0  someone shared with me that drone deliveries a...    not verified   
1  someone shared with me that there are more mic...    not verified   
2  someone shared with me that american industria...    not verified   
3  someone shared with me that the metro of st. p...    not verified   
4  someone shared with me that the number of busi...    not verified   

  author_ban_status  video_view_count  video_like_count  video_share_count  \
0      under review          343296.0           19425.0              241.0   
1            active          140877.0           77355.0            19034.0   
2            active          902185.0           97690.0             2858.0   
3            active          437506.0          239954.0            34812.0   
4            active           56167.0           34987.0             4110.0   

   video_download_count  video_comment_count  
0                   1.0                  0.0  
1                1161.0                684.0  
2                 833.0                329.0  
3                1234.0                584.0  
4                 547.0                152.0

You are interested in the relationship between `verified_status` and `video_view_count`. One approach is to examine the mean value of `video_view_count` for each group of `verified_status` in the sample data.

In [8]:
# Compute the mean `video_view_count` for each group in `verified_status`
data_cleaned.groupby('verified_status')['video_view_count'].mean()

verified_status
not verified    265663.785339
verified         91439.164167
Name: video_view_count, dtype: float64

### **Task 3. Hypothesis testing**

Before you conduct your hypothesis test, consider the following questions where applicable to complete your code response:

1. Recall the difference between the null hypothesis and the alternative hypotheses. What are your hypotheses for this data project?

**Answer:**  
null hypothesis = no difference between verified status and unverified status mean value for 'video_view_count.' Any difference is due to chance or sampling variability.

alt hypothesis = there is a statistically significant difference between verified and unverified mean value for 'video_view_count'



Your goal in this step is to conduct a two-sample t-test. Recall the steps for conducting a hypothesis test:


1.   State the null hypothesis and the alternative hypothesis
2.   Choose a signficance level
3.   Find the p-value
4.   Reject or fail to reject the null hypothesis



**Answser:**  
Null hypothesis: There is no difference in the mean number of views between videos posted by verified accounts and videos posted by unverified accounts. Any difference is due to chance or sampling variability.

Alternative hypothesis: There is a difference in the mean number of views between videos posted by verified accounts and videos posted by unverified accounts.

You choose 5% as the significance level and proceed with a two-sample t-test.

In [9]:
# Conduct a two-sample t-test to compare means

verified_sample = data_cleaned[(data_cleaned['verified_status']=='verified')]['video_view_count']
#print(verified_mean)

unverified_sample = data_cleaned[(data_cleaned['verified_status']=='not verified')]['video_view_count']
#print(unverified_mean)

stats.ttest_ind(a=verified_sample, b=unverified_sample, equal_var=False)

Ttest_indResult(statistic=-25.499441780633777, pvalue=2.6088823687177823e-120)

**Question:** Based on the p-value you got above, do you reject or fail to reject the null hypothesis?


**Answer:**  
The p-value is miniscule (e<sup>-120</sup>) which means it's under the 5% significance level. Since this is the case, there is a statiscally significant difference in the mean for videos posted by verified and unverified accounts so we reject the null hypothesis in favor of the alternative hypothesis. 

<img src="images/Execute.png" width="100" height="100" align=left>

## **PACE: Execute**

Consider the questions in your PACE Strategy Documentto reflect on the Execute stage.

## **Step 4: Communicate insights with stakeholders**

*Ask yourself the following questions:*

- What business insight(s) can you draw from the result of your hypothesis test?

**Answer:**  
There is a statiscally significant difference in the mean number of view counts for videos posted by verified and unverified accounts. This means that the difference is not caused by chance, rather it could be behavioral differences caused by click bait vidoes or bots that inflate view counts.

**Congratulations!** You've completed this lab. However, you may not notice a green check mark next to this item on Coursera's platform. Please continue your progress regardless of the check mark. Just click on the "save" icon at the top of this notebook to ensure your work has been logged.